### PR TITLE
Add step execution progress

### LINKED ISSUE #355: Refactor and port step execution progress view to Angular 4
As a developer, I'd like to refactor and port Step Executions progress view (including: steps, executions, and step execution widgets) to Angular 2.0 stack.

**Acceptance:**
- Tab ported over based on the standardized infrastructure 
- Data model between the UI and backend is consistent (since Typescript supports generics and casting like Java)
- Tab adapts to PUI theme
- Existing tests adapted to the new stack
- Test coverage improved

### PR DESCRIPTION
**Pull request by jvalkeal**
- Add step execution progress view and
  progress in step execution details.
- Add refresh and other navigations.
- Fixes #355

### CODE REVIEW

File: ui/src/app/jobs/jobs.service.ts [hunk 1]
```diff
@@ -80,6 +81,134 @@ export class JobsService {
...
+   * @param {string} jobid The job execution id for the step.
+   * @param {string} stepid The step execution id.
+   * @returns {Observable<StepExecutionProgress>}
+   */
+  getStepExecutionProgress(jobid: string, stepid: string): Observable<StepExecutionProgress> {
+    return this.http.get(this.jobExecutionsUrl + '/' + jobid + '/steps/' + stepid + '/progress', {})
+      .map(this.extractStepExecutionProgressData.bind(this))
+      .catch(this.errorHandler.handleError);
+  }
+  private extractStepExecutionProgressData(response: Response): StepExecutionProgress {
```
Comment (cppwfs) [inline ui/src/app/jobs/jobs.service.ts]: We need to add a test for this in the spec.  Codecov is yelling at us ;-)

Comment (cppwfs) [inline ui/src/app/jobs/jobs.service.ts]: Do you want me to create a story for this?

**Comment by cppwfs** [comment]
One question:  Do we want to wait for the results of your spike prior to merging?

**Comment by jvalkeal** [comment]
no, we don't need to wait that spike

File: ui/src/app/jobs/jobs.service.ts (hunk 1 above)
Comment (jvalkeal) [inline ui/src/app/jobs/jobs.service.ts]: Ah yes, forget to add test. We can do it with other story.

File: ui/src/app/jobs/jobs.service.ts [hunk 2]
```diff
@@ -80,6 +81,134 @@ export class JobsService {
...
+    const stepExecutionProgress: StepExecutionProgress = new StepExecutionProgress();
+    const body = response.json();
+    stepExecutionProgress.percentageComplete = body.percentageComplete;
+    stepExecutionProgress.finished = body.finished;
+    stepExecutionProgress.duration = body.duration;
+    const stepExecutionItem = body.stepExecution;
+    const stepExecution: StepExecution = new StepExecution();
```
Comment (jvalkeal) [inline ui/src/app/jobs/jobs.service.ts]: I'd prefer to keep it as it feels awkward not to map full response.

File: ui/src/app/jobs/model/step-execution-progress.model.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { StepExecution } from './step-execution.model';
+
+export class StepExecutionProgress {
```
Comment (jvalkeal) [inline ui/src/app/jobs/model/step-execution-progress.model.ts]: Ah yeah thx

File: ui/src/app/jobs/jobs.service.ts (hunk 1 above)
Comment (cppwfs) [inline ui/src/app/jobs/jobs.service.ts]: https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/365

File: ui/src/app/jobs/jobs.service.ts (hunk 2 above)
Comment (cppwfs) [inline ui/src/app/jobs/jobs.service.ts]: ok

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Opened Issue for testing, Added JSDocs, Rebased, Squashed Merged.

**Pull request closed without merging**